Common/Data/UniverseSelection/FuturesChainUniverse.cs

```diff
@@ -0,0 +1,196 @@
+
+        private readonly Future _future;
+        private readonly UniverseSettings _universeSettings;
+        private SubscriptionManager _subscriptionManager;
+
+        private HashSet<Symbol> _cachedSelection;
+        private DateTime _cacheDate;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="FuturesChainUniverse"/> class
+        /// </summary>
+        /// <param name="future">The canonical future chain security</param>
+        /// <param name="universeSettings">The universe settings to be used for new subscriptions</param>
+        /// <param name="subscriptionManager">The subscription manager used to return available data types</param>
+        /// <param name="securityInitializer">The security initializer to use on newly created securities</param>
+        public FuturesChainUniverse(Future future, 
+                                    UniverseSettings universeSettings,
+                                    SubscriptionManager subscriptionManager,
+                                    ISecurityInitializer securityInitializer = null)
+            : base(future.SubscriptionDataConfig, securityInitializer)
+        {
+            _future = future;
+            _universeSettings = universeSettings;
+            _subscriptionManager = subscriptionManager;
+        }
+
+        /// <summary>
+        /// Gets the settings used for subscriptons added for this universe
+        /// </summary>
+        public override UniverseSettings UniverseSettings
+        {
+            get { return _universeSettings; }
+        }
+
+        /// <summary>
+        /// Performs universe selection using the data specified
+        /// </summary>
+        /// <param name="utcTime">The current utc time</param>
+        /// <param name="data">The symbols to remain in the universe</param>
+        /// <returns>The data that passes the filter</returns>
+        public override IEnumerable<Symbol> SelectSymbols(DateTime utcTime, BaseDataCollection data)
+        {
+            var futuresUniverseDataCollection = data as FuturesChainUniverseDataCollection;
+            if (futuresUniverseDataCollection == null)
+            {
+                throw new ArgumentException(string.Format("Expected data of type '{0}'", typeof (FuturesChainUniverseDataCollection).Name));
+            }
+
+            var underlying = new Tick { Time = utcTime };
+
+            if (_cacheDate != null && _cacheDate == data.Time.Date)
+            {
+                return Unchanged;
+            }
+
+            var availableContracts = futuresUniverseDataCollection.Data.Select(x => x.Symbol);
+            var results = (FutureFilterUniverse)_future.ContractFilter.Filter(new FutureFilterUniverse(availableContracts, underlying));
+
+            // if results are not dynamic, we cache them and won't call filtering till the end of the day
+            if (!results.IsDynamic)
```

Comment:
Yes, it changes within the day. I have a comment about that in the interface definition. Can change the name if needed.